<launch>
  <group ns="pladypos">
    <rosparam command="load" file="$(find pladypos)/launch/model.yaml" />
    <rosparam command="load" file="$(find caddy_uwsim)/launch/simulation.yaml" />
  	
    <!-- <node pkg="cart_sim" name="radio_node" type="radio_node">
      <param name="PortName" value="/dev/ttyUSB0" />
      <param name="Baud" value="9600" />
      <remap from="joy_out" to="joy" />
    </node> -->
    
    <node pkg="pladypos" name="pladypos_node" type="pladypos_node">
      <param name="PortName" value="/dev/ttyS0" />
      <param name="Baud" value="57600" />
      <param name="maxThrust" value="1"/>
      <param name="minThrust" value="-1"/>
      <remap from="tauIn" to="tauOut" />
    </node> 
    
    <node pkg="labust_uvapp" name="velocity_control" type="velocity_control" />

    <node pkg="microstrain_3dmgx2_imu" type="imu_node" name="microstrain_3dmgx2_node" output="screen">
      <param name="time_offset" value="-0.040" />
      <param name="port"  type="string" value="/dev/ttyACM0" />
      <param name="autostart" type="bool" value="true" />
      <param name="frameid"  type="string" value="imu" />
      <param name="autocalibrate" type="bool" value="true" />
      <param name="angular_velocity_stdev" type="double" value="0.00017" />
    </node>

    <node pkg="pladypos" type="nav_node" name="nav_node">
      <remap from="imu" to="imu/data" />
    </node>
    
    <!-- <node pkg="gpsd_client" name="gpsd_client" type="gpsd_client" /> -->
  </group>
</launch>
